pkg/apis/navigator/v1alpha1/types.go

```diff
@@ -15,8 +15,9 @@ const (
 	ElasticsearchNodePoolVersionAnnotation = "navigator.jetstack.io/elasticsearch-version"
 	ElasticsearchRoleLabelPrefix           = "navigator.jetstack.io/elasticsearch-role-"
 
-	CassandraClusterNameLabel  = "navigator.jetstack.io/cassandra-cluster-name"
-	CassandraNodePoolNameLabel = "navigator.jetstack.io/cassandra-node-pool-name"
+	CassandraClusterNameLabel   = "navigator.jetstack.io/cassandra-cluster-name"
+	CassandraNodePoolNameLabel  = "navigator.jetstack.io/cassandra-node-pool-name"
+	CassandraNodePoolIndexLabel = "navigator.jetstack.io/cassandra-node-pool-index"
```

Comment:
This appears to be set, but not actually used (at least as part of this PR).

It also could introduce some issues around generalising this code between controllers if we ever have a controller that doesn't use a StatefulSet.

Can we remove it for now in favour of discussion at a later date of storing the index vs the pod name on the Pilot?